### PR TITLE
chore: add crash test

### PR DESCRIPTION
**Pull request by cutecutecat**
Part of https://github.com/tensorchord/pgvecto.rs/issues/215

This is a test of Crash Recovery.

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL @usamoi

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -87,7 +87,17 @@ jobs:
...
+          sqllogictest -u runner -d runner './tests/sqllogictest/*.slt'
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10' 
+      - name: Crash Recovery test
+        run: |
+          sudo pip install psutil==5.9.7
+          psql -f ./tests/init.sql
+          sqllogictest -u runner -d runner './tests/crash/create.slt'
+          sudo python ./tests/crash/kill.py
```
Comment (kemingy) [inline .github/workflows/check.yml:99]: Why does python/pip require `sudo`?

File: tests/crash/kill.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+def process_filter(p: psutil.Process):
+    cmdline = "".join(p.cmdline())
+    filter_func: Callable[[bool, str], bool] = lambda ans, e: ans and (e in cmdline)
+    return reduce(filter_func, AND_FILTERS, True)
+
+
+if __name__ == "__main__":
+    # Send kill signal to vectors process
+    timeout_start = time.time()
```
Comment (kemingy) [inline tests/crash/kill.py:21]: 1. use monotonic time
2. `+ TIMEOUT` here

File: tests/crash/kill.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+TIMEOUT = 30
+AND_FILTERS = ["postgres", "vectors"]
+logging.getLogger().setLevel(logging.INFO)
+
+
+def process_filter(p: psutil.Process):
+    cmdline = "".join(p.cmdline())
+    filter_func: Callable[[bool, str], bool] = lambda ans, e: ans and (e in cmdline)
+    return reduce(filter_func, AND_FILTERS, True)
```
Comment (kemingy) [inline tests/crash/kill.py:16]: This is a bad usage of `reduce`. Could you try the normal code?

File: tests/crash/kill.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        time.sleep(1)
+
+    # Wait until process is not exist or recreated
+    timeout_start = time.time()
+    while True:
+        if time.time() > timeout_start + TIMEOUT:
+            raise TimeoutError(f"Background worker not killed in {TIMEOUT}s")
+        procs = [p for p in psutil.process_iter() if process_filter(p)]
+        pids = set(p.pid for p in procs)
+        if len(procs) == 0 or pids != last_pids:
```
Comment (kemingy) [inline tests/crash/kill.py:41]: Do you mean `len(pids & last_pids) == 0`?

BTW, `last_pids` is kind of a temp var in the previous while loop.

File: .github/workflows/check.yml (hunk 1 above)
Comment (cutecutecat) [reply .github/workflows/check.yml:99]: Kill process need a priviledge, and sudo python would use a seperated interpreter from local user. I understand sudo is soso, if there is a better way?

Comment (kemingy) [reply .github/workflows/check.yml:99]: It's necessary if the process is started with `sudo`.

File: tests/crash/kill.py (hunk 2 above)
Comment (cutecutecat) [reply tests/crash/kill.py]: Fixed.

File: tests/crash/kill.py (hunk 3 above)
Comment (cutecutecat) [reply tests/crash/kill.py]: Fixed.

File: tests/crash/kill.py (hunk 4 above)
Comment (cutecutecat) [reply tests/crash/kill.py]: Fixed, now extend last_pids's lifetime.

**Review by kemingy — APPROVED** [review]

**Review by usamoi — APPROVED** [review]

**Pull request merged**